### DISCUSSION TITLE
Live update vscode interface when theme.json changes in production?

### DISCUSSION THREAD

**Discussion by sudoaugustin** [microsoft/vscode-discussions#31] [Extension Development QnA] [answered]
During developing a theme, when I change the `theme.json` file it updates the VSCode interface without needing to reload the window. Is there any way to do that to a production theme?

**Expected behavior: When I go and change the theme.json file (in production), it should update the VSCode color.
The reason I need this is I'm developing [VSLook](https://marketplace.visualstudio.com/items?itemName=sudoaugustin.vslook)**. For now, all theme changes are stored in `setting.json` and it's live updated.

**Comment by miguelsolorio** [comment]
Can you elaborate on what you mean by a "production theme"? Taking a look at the VSLook extension I wasn't able to understand what you meant. Is the theme stored externally and then downloaded locally when using the extension?

**Reply by sudoaugustin** [reply]
I found the way. Thx for the support

**Reply by miguelsolorio** [reply]
Can you add your solution so others that come across this can know how to resolve it?

**Reply by sudoaugustin** [reply]
Sorry I didn't find the solution. But I just found another way to do what I wanna do.


Basically what I want is I install OneDarkPro Theme. If I change the `theme.json` file of OneDarkPro, it has to update the VSCode colors without needing to reload the window. This feature is available in Debug VSCode windows.

**Reply by miguelsolorio** [reply]
Are you editing the theme source files from the extensions folder in your user directory or are you editing the project files? Based on the video below it looks like you are editing the theme source files, which is not recommended as that will be overwritten when there are extension updates. You should always edit the files in your project or use your user settings to scope theme changes via `workbench.colorCustomizations`

**Reply by sudoaugustin** [reply]
I store theme in a `globalState` and overwrite the `theme.json` file. Will  `globalState` be persisted across new install? Any commands or ways to refresh the UI?(Not window reload)
And Thx for answering. @misolori

**Comment by sudoaugustin** [comment, marked as answer]
In the video, the second window auto-refresh on the theme file change. But the first window stays the same until it's reopened.

https://user-images.githubusercontent.com/57541158/173178135-44c9862c-549f-4ac9-b52a-964f7a55712e.mov
